### PR TITLE
resolve merge conflict for PR #151 

### PR DESCRIPTION
**Pull request by faceyspacey**
This PR resolves: https://github.com/webpack/webpack-dev-middleware/pull/151

@iMoses 's original code worked fine, but just needed to be updated with other changes that occurred in the *webpack-dev-middleware* package over the last few months. All tests are passing. I also made the primary code more readable, i.e. the code to determine the `publicPath` and `outputPath`.

In addition, I've personally used it in both a multi-compiler app and a non-multi-compiler app, and it works just fine. It doesn't do that much. It shouldn't be a problem. Hopefully we can get this merged soon.

### CODE REVIEW

**Comment by shellscape** [comment]
thanks @faceyspacey, will have a look tomorrow

**Comment by shellscape** [comment]
@rainercedric23 @xonar @rfloriano please give this branch a try and let us know how it works for you

**Comment by faceyspacey** [comment]
any word on merging on this? Today I'm releasing another package that depends on this and will need to create an unnecessary package to depend on to make documentation straightforward.

**Comment by shellscape** [comment]
I'd like to wait for the folks mentioned to test this on their implementations. I completely understand that there's some urgency on your end, but I feel patience on this PR is prudent.

**Comment by faceyspacey** [comment]
Anyone had a chance to check this out?

**Comment by faceyspacey** [comment]
Anyone get a chance to look at this? I think what happened last time is it didn't get merged, and other commits came in, and this had to be updated. That seems to be what will happen again if nobody takes a look at this. @shellscape I don't suppose you can check it out?

**Comment by shellscape** [comment]
@faceyspacey kill that yarn.lock file in the commit and I'll merge it

**Comment by faceyspacey** [comment]
done.

File: README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@
   <h1>webpack Dev Middleware</h1>
 </div>
 
+**NOTE: This is a temporary package that adds support for multiple compilers with a pending PR that likely will be merged soon: https://github.com/webpack/webpack-dev-middleware/pull/187 ...don't sweat it.**
```
Comment (shellscape) [inline README.md]: looks like this snuck in on you

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "webpack-dev-middleware",
-  "version": "1.10.1",
+  "name": "webpack-dev-middleware-multi-compiler",
+  "version": "1.0.0",
```
Comment (shellscape) [inline package.json]: and this change in package.json too

Comment (faceyspacey) [inline package.json]: done, sorry about that.

**Comment by shellscape** [comment]
Thanks very much for putting the time into this. I'll get with the NPM owners and try to get this published timely.

**Comment by faceyspacey** [comment]
any chance this can get published to NPM soon?

**Comment by shellscape** [comment]
I've pinged @SpaceK33z

**Comment by 7rulnik** [comment]
@SpaceK33z sorry for disturbing, but can you release it?

**Comment by shellscape** [comment]
I've been given the ability to publish now 🎉  but I won't be able to roll this one out until tomorrow.

**Comment by 7rulnik** [comment]
It's fine. Can you ping me after release?

**Comment by eudaimos** [comment]
@shellscape was this published?

**Comment by shellscape** [comment]
yes, it's published. and no worries

**Pull request merged**